### PR TITLE
Add functionality on Clear completed button

### PR DESCRIPTION
**Pull request by Kamba56**
Added functionality to the checkbox
Clear list items that are completed
Updated the index of all list items

### CODE REVIEW

File: src/modules/populate.js [hunk 1]
```diff
@@ -48,8 +48,14 @@ const populate = (todo) => {
...
   });
 
+  if (todo.completed) {
+    checkbox.checked = true;
+    text.style.textDecoration = 'line-through';
+  }
+
   checkbox.addEventListener('change', (e) => {
     complete(e, listelement, text);
+    localStorage.setItem('list', JSON.stringify(list));
```
Comment (Cathella) [inline src/modules/populate.js:58]: ✅ It has event listener on the checkbox (change). 👍🏻

File: src/index.js [hunk 2]
```diff
@@ -16,3 +18,26 @@ for (let i = 1; i <= list.length; i += 1) {
...
+  const newlist = list.filter((element) => element.completed === false);
+  list.forEach((element) => {
+    if (!newlist.includes(element)) {
+      list.splice(list.indexOf(element), 1);
+    }
+  });
+  let i = 1;
+  list.forEach((element) => {
+    element.index = i;
+    i += 1;
```
Comment (Cathella) [inline src/index.js:32]: ✅ A function for the "Clear all completed" button using the Array.filter() method is implemented 👍🏻 

- [ ] [REQUIRED] The function does not work well yet. When I check all the tasks to be cleared, some are left in the list which shouldn't be the case.

| Before | After |
| --- | --- |
| image | image |

**Review by Cathella — CHANGES_REQUESTED** [review]
## 🟡 Status: Changes Required 🛑

### Hi @Kamba56 👋🏻, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## To Highlight 👏🏻

✅ No linter errors.
✅ You used correct GitHub Flow.
✅ The task object's values for completed is updated upon user actions.

## Required Changes ♻️

- [ ] Create a new JavaScript file imported as a module to contain methods related to the status updates (completed: true / false).

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kamba56** [comment]
I have a js file imported as a module that implements the completed = true or false. The file is check.js located in my modules folder. @Cathella

**Review by roshan-bajgain — APPROVED** [review]
Hi @Kamba56, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opi

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**